### PR TITLE
cmd/cue/cmd: fix bug with help flag

### PR DESCRIPTION
**Pull request by verdverm**
Fixes https://github.com/cuelang/cue/issues/340 

- [x] add more tests for help system
- [x] find and fix bug

### CODE REVIEW

**Comment by verdverm** [comment]
It's sort of weird that the `cue cmd --help` does not behave like the other subcommands, but it is different from the others and how it is used.

File: cmd/cue/cmd/root.go [hunk 1]
```diff
@@ -234,7 +234,9 @@ func New(args []string) (cmd *Command, err error) {
 		// "fix":   {"fix", nil},
 	}
 
-	if args[0] == "help" {
+	fmt.Println("GOT HERE - pre subcmds", args)
```
Comment (mpvl) [inline cmd/cue/cmd/root.go:237]: remove

File: cmd/cue/cmd/root.go [hunk 2]
```diff
@@ -244,12 +246,17 @@ func New(args []string) (cmd *Command, err error) {
 		return cmd, addSubcommands(cmd, sub, args, false)
 	}
 
+	fmt.Println("GOT HERE - pre flags")
```
Comment (mpvl) [inline cmd/cue/cmd/root.go:249]: remove

File: cmd/cue/cmd/root.go [hunk 3]
```diff
@@ -244,12 +246,17 @@ func New(args []string) (cmd *Command, err error) {
...
+
 	// TODO: clean this up once we either use Cobra properly or when we remove
 	// it.
 	err = cmd.cmd.ParseFlags(args)
 	if err != nil {
 		return nil, err
 	}
+
+	fmt.Println("GOT HERE - post flags")
```
Comment (mpvl) [inline cmd/cue/cmd/root.go:258]: remove

File: cmd/cue/cmd/root_test.go [hunk 4]
```diff
@@ -17,8 +17,35 @@ package cmd
 import "testing"
 
 func TestHelp(t *testing.T) {
-	cmd, err := New([]string{"help", "cmd"})
+
```
Comment (mpvl) [inline cmd/cue/cmd/root_test.go:20]: remove blank line

Comment (verdverm) [reply cmd/cue/cmd/root_test.go:20]: @mpvl would you like all of the blank lines removed from that file?

**Comment by verdverm** [comment]
I have a changeset on Gerrit with the fixes needed here, so I updated GitHub with a force and abandoned the Gerrit CL.

**Comment by mpvl** [comment]
@verdverm Ah, I didn't notice the Gerrit CL.

File: cmd/cue/cmd/root_test.go (hunk 4 above)
Comment (mpvl) [reply cmd/cue/cmd/root_test.go:20]: No just as the first line of a function. Blanks between blocks of code is fine and encouraged even.

**Comment by verdverm** [comment]
@mpvl I'm all setup with Gerrit, so I'll be doing that going forward.

**Pull request closed without merging**